core/network/impl/synchronizer_impl.cpp

```diff
@@ -627,7 +627,8 @@ namespace kagome::network {
           return;
         }
         // Check if body is provided
-        if (need_body and not block.body.has_value()) {
+        if (need_body and block.header->number != 0
```

Comment:
would be great to somehow use `[unlikely]` attribute for the header number check. But up to you